src/events/SDL_events.c

```diff
@@ -952,17 +976,32 @@ SDL_WaitEventTimeout(SDL_Event * event, int timeout)
     SDL_Window *wakeup_window;
     Uint32 start = 0;
     Uint32 expiration = 0;
+    SDL_bool include_sentinel = (timeout == 0) ? SDL_TRUE : SDL_FALSE;
```

Comment:
Hmm, don't we want to retry (without pumping) if we encounter the sentinel for all cases?

Reply:
No, we want to skip the sentinel when returning from SDL_PeepEvents() to application code. The only time we want to see it is when we want to stop a SDL_PollEvent() cycle.

Reply:
include_sentinel == SDL_FALSE is equivalent to retrying without pumping, but more efficient, since it's handled inside SDL_PeepEvents()

Reply:
Ah yep, got it now.